backend/api/views/dataset.py

```diff
@@ -84,4 +86,17 @@ def datawork_import(self, request):
         queryset = new_datasets.annotate(Count("files")).select_related("dataset_type")
         serializer = self.serializer_class(queryset, many=True)
 
+        try:
+            check_new_spectro_config()
```

Comment:
So you only use this to get the error ? Why does it have a return ? Will it be used in other way later on ? Otherwise just make it return a boolean and call it `check_new_spectro_config_validity` and do the except there.

Reply:
Okay sorry if this wasn't clear. I don't like a function meant to error but also sending different returns isn't ideal : the good solution is `errors = check_new_spectro_config_errors()` that returns errors messsages if there are errors (try and except is in that other function/file, not here) and if errors is empty then no need to do anything.